docs/examples/auth.py

```diff
@@ -0,0 +1,70 @@
+import jwt
+
+from nameko.extensions import DependencyProvider
+
+
+JWT_SECRET = "secret"
+
+
+class Unauthenticated(Exception):
+    pass
+
+
+class Unauthorized(Exception):
+    pass
+
+
+class Auth(DependencyProvider):
+
+    class Api:
+        def __init__(self, users, worker_ctx):
+            self.users = users
+            self.worker_ctx = worker_ctx
+
+        def authenticate(self, username, password):
+            try:
+                assert self.users[username]['password'] == password
```

Comment:
i know this is just a toy but i'm still not sure i'm comfortable with example code suggesting stored plain text passwords